src/ck-core/ck.C

```diff
@@ -548,12 +548,12 @@ void CkDeliverMessageFree(int epIdx,void *msg,void *obj)
 #if CMK_CHARMDEBUG
   CpdBeforeEp(epIdx, obj, msg);
 #endif    
-  const auto msgtype = UsrToEnv(msg)->getMsgtype();
+  const auto msgtype = (msg == NULL) ? LAST_CK_ENVELOPE_TYPE : UsrToEnv(msg)->getMsgtype();
```

Comment:
Why is it now possible for NULL messages to be coming through this code path?

Reply:
I am seeing them come through during migration, for a constructor of an array element which doesn't take a message. This was identified in charm's megatest after the first commit of this patch.